internal/component/prometheus/relabel/relabel.go

```diff
@@ -259,6 +272,9 @@ func (c *Component) relabel(val float64, lbls labels.Labels) labels.Labels {
 	// Set the cache size to the cache.len
 	// TODO(@mattdurham): Instead of setting this each time could collect on demand for better performance.
 	c.cacheSize.Set(float64(c.cache.Len()))
+
+	c.debuggingStreamHandler.Stream(c.opts.ID, fmt.Sprintf("%s => %s", lbls.String(), relabelled.String()))
```

Comment:
Yeah it will, that's a good remark because before I had a check to see whether there was a stream or not and I removed it to reduce the verbosity. Maybe it's better to put the check back to avoid computing the string (the check just checks the map if a stream exists or not, it's very light)

Reply:
+1 `.String()` will be expensive and I've actually worked on a fix in the past that had this very issue when doing debug statements for the logger

We should make it lazy in some way. Logging libraries can be used as an inspiration on how this can be achieved.